web/src/test/java/org/springframework/security/web/server/util/matcher/IpAddressServerWebExchangeMatcherTests.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2002-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.server.util.matcher;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import org.springframework.mock.http.server.reactive.MockServerHttpRequest;
+import org.springframework.mock.web.server.MockServerWebExchange;
+import org.springframework.web.server.ServerWebExchange;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+
+/**
+ * @author Guirong Hu
+ * @since 6.0
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class IpAddressServerWebExchangeMatcherTests {
+
+	final IpAddressServerWebExchangeMatcher v4matcher = new IpAddressServerWebExchangeMatcher("192.168.1.104");
+
+	final IpAddressServerWebExchangeMatcher v6matcher = new IpAddressServerWebExchangeMatcher(
```

Comment:
Some of the tests already require a new instance of the matcher inline. I'd suggest moving both v4 and v6 matchers to the necessary test methods, as this small form of duplication is preferred here.

Reply:
Done.